### PR TITLE
Plexo: Add new gateway

### PR DESCRIPTION
**Pull request by ajawadmirza**
Added plexo gateway implementation along with remote and unit test
cases.

CE-2536

Unit:
5201 tests, 75845 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Rubocop:
742 files inspected, no offenses detected

Remote:
15 tests, 43 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

File: test/remote/gateways/remote_plexo_test.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  def test_failed_void
+    response = @gateway.void('123', @cancel_options)
+    assert_failure response
+    assert_equal 'An internal error occurred. Contact support.', response.message
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@credit_card, @options.merge(@cancel_options))
+    assert_success response
+    assert_match %r{You have been mocked.}, response.message
```
Comment (therufs) [inline test/remote/gateways/remote_plexo_test.rb:124]: Is there a reason to switch from `assert_equal` to `assert_match` here? I think either would be fine, but am wondering why there are some of each.

File: lib/active_merchant/billing/gateways/plexo.rb [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      def initialize(options = {})
+        requires!(options, :client_id, :api_key)
+        @credentials = options
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        response = MultiResponse.run do |r|
+          r.process { authorize(money, payment, options) }
+          r.process { capture(money, r.authorization, options) }
```
Comment (therufs) [inline lib/active_merchant/billing/gateways/plexo.rb:25]: What happens here if the `authorize` fails?

**Review by therufs — COMMENTED** [review]
Hey @ajawadmirza -- I had a couple of questions, but this looks good!

File: test/remote/gateways/remote_plexo_test.rb (hunk 1 above)
Comment (ajawadmirza) [reply test/remote/gateways/remote_plexo_test.rb]: You're right, I have changed it to `assert_equal` to keep it consistent with other tests as there is no any specific reason to use that.

File: lib/active_merchant/billing/gateways/plexo.rb (hunk 2 above)
Comment (ajawadmirza) [reply lib/active_merchant/billing/gateways/plexo.rb:25]: It should be able to respond with error as expected. Here is the test to cover that case:

https://github.com/activemerchant/active_merchant/pull/4432/files#diff-b724dc322c2cc0818829951d8a93bb9fddbffd55fd3a717be266ecffe9d000a6R44

Comment (therufs) [reply lib/active_merchant/billing/gateways/plexo.rb:25]: Great, thanks! I went looking for the test and just didn't find it.

**Review by therufs — APPROVED** [review]
tests pass && lgtm!

**Pull request merged**